lib/Dialect/TensorExt/Transforms/RotateAndReduce.cpp

```diff
@@ -54,6 +55,9 @@ struct RotateAndReduce : impl::RotateAndReduceBase<RotateAndReduce> {
           llvm::TypeSwitch<Operation *, LogicalResult>(upstreamOpPtr)
               .Case<arith::ConstantOp>(
                   [&](auto upstreamOp) { return success(); })
+              // Ignore generic ops
+              .template Case<secret::GenericOp>(
+                  [&](auto upstreamOp) { return success(); })
```

Comment:
What's going on with the explicit handling of  `secret.generic` op here? I'm guessing it's because it can be a parent op?What about other "container" ops that provide block arguments (e.g. `affine.for`) then?